independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/jaxrs/ResponseImpl.java

```diff
@@ -283,22 +283,22 @@ public Builder getLinkBuilder(String relation) {
 
     @Override
     public MultivaluedMap<String, String> getStringHeaders() {
-        // FIXME: is this mutable?
         if (stringHeaders == null) {
             // let's keep this map case-insensitive
             stringHeaders = new CaseInsensitiveMap<>();
-            for (Entry<String, List<Object>> entry : headers.entrySet()) {
-                List<String> stringValues = new ArrayList<>(entry.getValue().size());
-                for (Object value : entry.getValue()) {
-                    stringValues.add(HeaderUtil.headerToString(value));
-                }
-                stringHeaders.put(entry.getKey(), stringValues);
-            }
+            headers.forEach(this::populateStringHeaders);
         }
-
         return stringHeaders;
     }
 
+    public void populateStringHeaders(String headerName, List<Object> values) {
+        List<String> stringValues = new ArrayList<>(values.size());
```

Comment:
I know it's not related to this PR, but allocating arrayslists in a tight loop for no other reason than to accomodate for the `put` method requirements ?! :)

Reply:
I'm open to suggestions :)